File: solutions/go/parsing-log-files/2/parsing_log_files.go
package parsinglogfiles

import (
    "regexp"
    "fmt"
)

var regexmap map[string]*regexp.Regexp

func init() {
    regexmap = make(map[string]*regexp.Regexp)
    regexmap["IsValidLine"] = regexp.MustCompile(`^\[TRC\]|^\[DBG\]|^\[INF\]|^\[WRN\]|^\[ERR\]|^\[FTL\]`)
    regexmap["SplitLogLine"] = regexp.MustCompile(`<[~*=-]*>`)
    regexmap["CountQuotedPasswords"] = regexp.MustCompile(`(?i)".*password.*"`)
    regexmap["RemoveEndOfLineText"] = regexp.MustCompile(`end-of-line[\d]+`)
    regexmap["TagWithUserName"] = regexp.MustCompile(`User[\s]+(\S*)\b`)
}

func IsValidLine(text string) bool {
    return regexmap["IsValidLine"].MatchString(text)
}

func SplitLogLine(text string) []string {
    return regexmap["SplitLogLine"].Split(text, -1)
}

func CountQuotedPasswords(lines []string) int {
    cnt := 0
    for _, v := range lines {
        if regexmap["CountQuotedPasswords"].MatchString(v) {
            cnt++
        }
    }
    return cnt
}

func RemoveEndOfLineText(text string) string {
    return regexmap["RemoveEndOfLineText"].ReplaceAllString(text, "")
}

func TagWithUserName(lines []string) []string {
    res := make([]string, 0)
    for _, v := range lines {
        mat := regexmap["TagWithUserName"].FindStringSubmatch(v)
        if len(mat) > 1 {
            res = append(res, fmt.Sprintf("[USR] %v %v", mat[1], v))
        } else {
            res = append(res, v)
        }
    }
    return res
}

File: solutions/csharp/parsing-log-files/2/ParsingLogFiles.cs
using System.Text.RegularExpressions;

public class LogParser
{
    private readonly Regex validLine = new(@"^\[(TRC|DBG|INF|WRN|ERR|FTL)\]");
    public bool IsValidLine(string text) => validLine.IsMatch(text);

    private readonly Regex delimiter = new(@"<[\^*=-]+>");
    public string[] SplitLogLine(string text) => delimiter.Split(text);

    private readonly Regex password = new(@""".*password.*""", RegexOptions.IgnoreCase);
    public int CountQuotedPasswords(string lines) => lines.Split(Environment.NewLine)
        .Count(line => password.IsMatch(line));

    private readonly Regex eol = new(@"end-of-line[0-9]+");
    public string RemoveEndOfLineText(string line) => eol.Replace(line, "");

    private readonly Regex offending = new(@"\bpassword\S+", RegexOptions.IgnoreCase);
    public string[] ListLinesWithPasswords(string[] lines) => lines.Select(line =>
        {
            Match match = offending.Match(line);
            return match.Success ? $"{match.Value}: {line}" : $"--------: {line}";
        }).ToArray();
}